src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -1733,6 +1736,48 @@ public void testRelativizedFileExclusion() throws Exception {
                 tempFile.getName(), expected);
     }
 
+    @Nested
+    public class TestProcessPathCollection {
+
+        @Test
+        public void testFileExtensionsPath() throws Exception {
```

Comment:
why we need nested? 
lets keep simple test methods

Reply:
My intent was to have this in a dedicated place, as opposed to being removed/merged afterward, at which point the new nesting would be gone. This is intended for when the test is fully migrated to new methods and the deprecated method is no longer used in subsequent pull requests.

I've migrated all the tests in a dedicated branch, and this was a significant amount of work considering the file mocking.

However, inlining it and removing/adapting afterward would have been the same amount of effort.

Will apply.